File: ios/Sources/TextConverterPlugin/TextConverterPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(TextConverterPlugin)
public class TextConverterPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "TextConverterPlugin"
    public let jsName = "TextConverter"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "convert", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = TextConverter()

    @objc func convert(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.convert(value)
        ])
    }
}

File: android/src/main/java/jp/newaid/textconverter/TextConverterPlugin.java
package jp.newaid.textconverter;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "TextConverter")
public class TextConverterPlugin extends Plugin {

    private TextConverter implementation = new TextConverter();

    @PluginMethod
    public void convert(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.convert(value));
        call.resolve(ret);
    }
}
